aws/rust-runtime/aws-sigv4/src/http_request/mod.rs

```diff
@@ -4,594 +4,53 @@
  */
 
 //! Utilities to sign HTTP requests.
-
-use crate::http_request::canonical_request::{CanonicalRequest, StringToSign};
-use crate::sign::{calculate_signature, generate_signing_key, sha256_hex_string};
-use crate::SigningOutput;
-use http::header::{HeaderName, HeaderValue};
-use std::error::Error as StdError;
-use std::{iter, str};
+//!
+//! # Example: Signing an HTTP request
+//!
+//! ```rust
+//! # fn test() -> Result<(), aws_sigv4::http_request::Error> {
+//! use aws_sigv4::http_request::{sign, SigningSettings, SigningParams, SignableRequest};
+//! use chrono::Utc;
+//! use http;
+//!
+//! // Create the request to sign
+//! let mut request = http::Request::builder()
+//!     .uri("https://some-endpoint.some-region.amazonaws.com")
+//!     .body("")
+//!     .unwrap();
+//!
+//! // Set up information and settings for the signing
+//! let signing_settings = SigningSettings::default();
+//! let signing_params = SigningParams::builder()
+//!     .access_key("example access key")
+//!     .secret_key("example secret key")
+//!     .region("us-east-1")
+//!     .service_name("exampleservice")
+//!     .date_time(Utc::now())
+//!     .settings(signing_settings)
+//!     .build()
+//!     .unwrap();
+//! // Convert the HTTP request into a signable request
+//! let signable_request = SignableRequest::from_http(&request);
```

Comment:
should this just be a `From<&'a http::Request>`?

Reply:
Huh. I had tried this and ran into a lifetime issue that I'm now unable to replicate. Will switch to implementing `From`.